exp/metadataserver/server_test.go

```diff
@@ -0,0 +1,133 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package metadataserver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	enterrors "github.com/weaviate/weaviate/entities/errors"
+	"github.com/weaviate/weaviate/exp/metadataserver/proto/api"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// TODO add tests for shutting down metadata/core/querier nodes and make sure they work as expected
+
+// testServerSetup gives you a ready to use Server and Logger and tells you which port
+// number to use.
+func testServerSetup(t *testing.T) (*Server, *logrus.Logger, int) {
+	log := logrus.New()
+	log.SetLevel(logrus.ErrorLevel)
+	// TODO replace real port with bufconn? This causes a firewall alert to trigger
```

Comment:
this and other non-critical TODOs will be addressed in follow up PRs, ran out of time to include them in this PR